docs/tutorials/morpheusvm/intro.md

```diff
@@ -0,0 +1,766 @@
+
+This function returns a slice of `uint16` values (one for each state key) and they
+both update a single balance key-value pair, so we'll use `BalanceChunks` as before.
+
+Let's update the function like so:
+
+```golang
+func (*Transfer) StateKeysMaxChunks() []uint16 {
+	return []uint16{BalanceChunks, BalanceChunks}
+}
+```
+
+### `Execute()`
+
+We start off by first defining the values necessary for this function at the
+top of the file:
+
+```golang
+const MaxMemoSize = 256
+
+var (
+	ErrOutputValueZero    = errors.New("value is zero")
+	ErrOutputMemoTooLarge = errors.New("memo is too large")
+)
+```
+
+We can now define `Execute()`. Recall that we should first check any invariants
+and then execute the necessary state changes. Therefore, we have the following:
+
+```golang
+func (t *Transfer) Execute(
+	ctx context.Context,
+	_ chain.Rules,
+	mu state.Mutable,
+	_ int64,
+	actor codec.Address,
+	_ ids.ID,
+) ([][]byte, error) {
+	if t.Value == 0 {
+		return nil, ErrOutputValueZero
+	}
+	if len(t.Memo) > MaxMemoSize {
+		return nil, ErrOutputMemoTooLarge
+	}
+	if err := SubBalance(ctx, mu, actor, t.Value); err != nil {
+		return nil, err
+	}
+	if err := AddBalance(ctx, mu, t.To, t.Value, true); err != nil {
+		return nil, err
+	}
+	return nil, nil
+}
+```
+
+### ComputeUnits
+
+The `ComputeUnits` function specifies how many units of computation are consumed
+by this action. The HyperSDK uses dynamic, multi-dimensional fees and specifies its
+own resource limits ie. units of compute, storage, and bandwidth. This means it's 
+more important that `ComputeUnits` of different actions is proportional to the maximum,
+target, and the cost of other actions than it is to be exact. So, we'll simply return 1
+for `Transfer`.
+
+Let's define a constant, `TransferComputeUnits`, at the top of the file.
+
+```golang
+const TransferComputeUnits = 1
+```
+
+Now we can implement our function:
+
+```golang
+func (*Transfer) ComputeUnits(chain.Rules) uint64 {
+	return TransferComputeUnits
+}
+```
+
+### ValidRange
+
+Finally, we need to specify the range of timestamps where this action will be considered
+valid. We'll use `-1` to denote that there's no bound ie. `-1, -1` indicates that the action
+is valid forever.
+
+Now, we'll update our function stub:
+
+```golang
+func (*Transfer) ValidRange(chain.Rules) (int64, int64) {
+	return -1, -1
+}
+```
+
+## Bringing Everything Together
+
+So far, we’ve implemented the following:
+
+- Actions
+- Storage
+
+We can now finish this MorpheusVM tutorial by implementing the vm interface.
+Let's create the file `vm.go`. Here, we first define the “registry” of MorpheusVM:
+
+```golang
+package tutorial
+
+import (
+   "github.com/ava-labs/avalanchego/utils/wrappers"
+
+   "github.com/ava-labs/hypersdk/auth"
+   "github.com/ava-labs/hypersdk/chain"
+   "github.com/ava-labs/hypersdk/codec"
+   "github.com/ava-labs/hypersdk/genesis"
+   "github.com/ava-labs/hypersdk/vm"
+   "github.com/ava-labs/hypersdk/vm/defaultvm"
```

Comment:
On my local machine, I get the following:

```
error while importing github.com/ava-labs/hypersdk/vm/defaultvm: missing go.sum entry for module providing package github.com/gorilla/websocket (imported by github.com/ava-labs/hypersdk/api/ws); to add:
	go get github.com/ava-labs/hypersdk/api/ws@v0.0.16compiler
```

When running `go mod tidy` though, this fixes the above.